### ISSUE TITLE
Remove `local_world_size` and `node_rank` requirement from file name formatting

### ISSUE THREAD

**Issue by ilee300a** [mosaicml/composer#1511]
## 🚀 Feature Request
It would be more flexible to not have the `local_world_size` and `node_rank` requirement in file name formatting when running composer trainers.

## Motivation
When using other plug-ins, PyTorch does not necessarily set those variables and it is difficult/unnecessary to set those variables. Moreover, the default file format string does not even use those, so having to set those environmental variables seems like an extra hassle.

**Comment by xwjiang2010** [comment]
To provide some context, we are trying to integrate Mosaic with Ray AIR (Ray AI runtime). Each Ray training worker currently does not have the knowledge of `local_world_size` and `node_rank`. We wonder if those arguments can be dropped.

**Comment by hanlint** [comment]
Hi @xwjiang2010 and @ilee300a , thanks for the interest! We'll check internally to see whether these can be removed as a requirement. 

We typically use these to determine local node groups that share a filesystem (e.g. for streaming data) and fast NVLINK interconnects (e.g. for syncing batch norm within a local group).

I'm less familiar with Ray, for distributed ML training, is there a concept of groups of nodes that share filesystems? If not, and each work is its "own" node, you could just set `LOCAL_WORLD_SIZE=1` and `NODE_RANK=RANK`.

**Comment by xwjiang2010** [comment]
@hanlint 
Thanks for the response!
Ray AIR (Ray AI runtime) is using Ray primitives (remote actors/tasks) to do distributed ML. In Ray cluster, there is a concept of distributed dataset, that is accessible to all the training workers through Ray object store. This is preferable when data is bigger than what can be accommodated by single node. 

Ray AIR uses PlacementGroup to denote placement strategy for training workers (there are SPREAD, PACKED, STRICT_SPREAD, STRICT_PACKED etc). By default, the flavor is PACKED. If the node has multiple GPUs and NVLINK available, then that is leveraged for syncing. However, Ray AIR currently doesn't have a notion of distinguishing a sub group of workers on one GPU (LOCAL_WORLD).

It seems that Mosaic uses LOCAL_WORLD, NODE_RANK information beyond just logging (which is my impression cc @ilee300a ). Is that some Composer algorithms also leverage this information for fast syncing?

The general idea of integrating Ray AIR with Mosaic is to see if we could open up another dimension (cluster dimension) in addition to algorithmic enhancement. Could we get connected with the team if possible to look at some possibilities? Irene may also have some more detailed questions.

**Comment by hanlint** [comment]
Hi @xwjiang2010 thanks for the detailed explanation. I think a quick chat would be helpful here so we can enable Composer on Ray AIR. Could you send me an email at hanlin@mosaicml.com?
